src/System/Taffybar/ToggleMonitor.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE OverloadedStrings #-}
+module System.Taffybar.ToggleMonitor (
+  handleToggleRequests,
+  toggleableMonitors,
+  withToggleServer
+) where
+
+import           Control.Concurrent
+import qualified Control.Concurrent.MVar as MV
+import           Control.Monad
+import           Control.Monad.Trans
+import           Control.Monad.Trans.Maybe
+import qualified Data.Map as M
+import           Data.Maybe
+import qualified Graphics.UI.Gtk as Gtk
+import           Graphics.UI.Gtk.Gdk.Screen
+import           System.Taffybar
+import           Text.Read hiding (get, lift)
+import           Web.Scotty
+import           XMonad.Core ( whenJust )
+
+toggleableMonitors :: MV.MVar (M.Map Int Bool)
```

Comment:
Having a M.Map Int Bool just makes it easier (or more natural) to change the 'default value' when the bar hasn't yet been toggled on a particular monitor.

It also makes the code that toggles have fewer branches since you don't have to worry about whether you are doing an insert/delete operation, you just negate whatever the current value is.

``` haskell
toggleTaffyOnMon fn mon = do
        MV.modifyMVar_ enabledVar $ \numToEnabled -> do
          let current = fromMaybe True $ M.lookup mon numToEnabled
          return $ M.insert mon (fn current) numToEnabled
```

vs

``` haskell
toggleTaffyOnMon fn mon = do
        MV.modifyMVar_ enabledVar $ \enabledMonitors -> do
          let current = elem mon enabledMonitors
          return $ if current
                   then delete mon enabledMonitors
                   else insert mon enabledMonitors
```

All that said, I really don't have strong feelings about this and I'd be happy to change it in the dbus version that I'm about to write if you would prefer